pandas/tests/indexing/test_iloc.py

```diff
@@ -856,11 +865,21 @@ def test_series_indexing_zerodim_np_array(self):
         result = s.iloc[np.array(0)]
         assert result == 1
 
-    @pytest.mark.xfail(reason="https://github.com/pandas-dev/pandas/issues/33457")
-    def test_iloc_setitem_categorical_updates_inplace(self):
+    def test_iloc_setitem_categorical_updates_inplace(
+        self, using_array_manager, request
+    ):
+        # GH#35417
         # Mixed dtype ensures we go through take_split_path in setitem_with_indexer
+        if using_array_manager:
+            mark = pytest.mark.xfail(
+                reason="https://github.com/pandas-dev/pandas/issues/33457"
+            )
+            request.node.add_marker(mark)
+
         cat = Categorical(["A", "B", "C"])
-        df = DataFrame({1: cat, 2: [1, 2, 3]})
+        df = DataFrame({1: cat, 2: [1, 2, 3]}, copy=False)
```

Comment:
Is this change related to this PR? (as this seems a test about iloc, not setitem) Or is it this `copy=False` addition that fixes it?

Reply:
good catch, will split this off to a different PR